src/login.rs

```diff
@@ -73,3 +73,28 @@ pub fn get_cgroup(pid: Option<pid_t>) -> Result<String> {
     let cg = unsafe { MString::from_raw(c_cgroup) };
     Ok(cg.unwrap().to_string())
 }
+
+/// Determines the session identifier of a process.
+///
+/// Specific processes can be optionally targeted via their PID. When no PID is
+/// specified, operation is executed for the calling process.
+/// This method can be used to retrieve a session identifier.
+pub fn get_session(pid: Option<pid_t>) -> Result<String> {
+    let mut c_session: *mut c_char = ptr::null_mut();
+    let p: pid_t = pid.unwrap_or(0);
```

Comment:
I've seen optional pids used in other proc-related libraries, like [getpgid](https://docs.rs/nix/0.8.1/nix/unistd/fn.getpgid.html) in nix crate. OTOH I've also seen this 0-as-self in some libs, as a well as some `_self` suffixed methods like in [procinfo](https://docs.rs/procinfo/0.3.1/procinfo/pid/index.html) crate.

I'm probably biased because I introduced the signature above, but I *strongly* prefer the `Option<pid_t>` way because:
 * it makes the semantic explicit via language types, instead of relying on docstring and programmer background.
 * it makes pid handling uniform. Not everywhere pid=0 means self, eg `getppid() => 0` has some vastly different meaning.
 * it feels much more idiomatic. In C you are forced into this "overload 0 and negative cases for special semantics" because the type system has no safe tools for optional/default values.